app/common/src/badge/VerificationBadge.js

```diff
@@ -22,36 +33,47 @@ type Props = {|
  * | UNVERIFIED with very long text to make it clear |
  * `-------------------------------------------------`
  */
-export default function VerificationBadge({ verified }: Props) {
-  let styleSet = unverifiedStyle;
-  let badgeText = 'UNVERIFIED';
+export default class VerificationBadge extends React.Component<Props> {
+  viewContainerReference: React.Ref<typeof View> | null;
+
+  /**
+   * React will call the 'ref' callback with the DOM element when the
+   * component mounts, and call it with null when it unmounts.
+   * 'ref' callbacks are invoked before componentDidMount or
+   * componentDidUpdate lifecycle hooks.
+   */
+  storeContainerReference = (reference: React.Ref<typeof View> | null) =>
+    (this.viewContainerReference = reference);
+
+  recalculateWidth = (textElement: TextOnLayout) => {
+    if (!this.viewContainerReference) {
+      return;
+    }
+    // $FlowFixMe: FLow is broken for Native, this is correct: http://facebook.github.io/react-native/releases/0.49/docs/direct-manipulation.html
+    this.viewContainerReference.setNativeProps({
```

Comment:
I really don't know how to solve this better without redefining React.Ref<>...

Reply:
I think it's quite usual in React Native and the official type works with React but not with RN (React doesn't have `setNativeProps`).